disruption_py/inout/mds.py

```diff
@@ -52,13 +50,43 @@ def get_shot_connection(self, shot_id: int):
         return MDSConnection(self.conn, shot_id)
 
 
+def _better_mds_exceptions(func):
+    """
+    Decorator to catch MDSplus exceptions and recreate them with a more descriptive
+    error message which includes the tree and the node path.
+    """
+
+    def wrapper(*args, **kwargs):
+        self = args[0]
+        if len(args) > 1:
+            path = args[1]
+        else:
+            path = kwargs.get("path", None)
+        try:
+            return func(*args, **kwargs)
+        except MDSplus.mdsExceptions.TreeFOPENR:
+            nick = kwargs["tree_name"]
+            tree = self.get_tree_name_of_nickname(nick)
+            raise MDSplus.mdsExceptions.TreeFOPENR(
+                f"Tree not found. Nick: {nick}, Tree: {tree}"
```

Comment:
I thought for this specific exception that the nickname of the tree would have been useful, LMK otherwise.